src/playlist-loader/dash-media-playlist-loader.js

```diff
@@ -0,0 +1,235 @@
+import containerRequest from '../util/container-request.js';
+import {addSidxSegmentsToPlaylist} from 'mpd-parser';
+import parseSidx from 'mux.js/lib/tools/parse-sidx';
+import {toUint8} from '@videojs/vhs-utils/es/byte-helpers';
+import {segmentXhrHeaders} from '../xhr';
+import {mergeMedia, forEachPlaylist} from './utils.js';
+
+/**
+ * This function is used internally to keep DashMainPlaylistLoader
+ * up to date with changes from this playlist loader.
+ *
+ * @param {Object} mainManifest
+ *        The manifest from DashMainPlaylistLoader
+ *
+ * @param {string} uri
+ *        The uri of the playlist to find
+ *
+ * @return {null|Object}
+ *         An object with get/set functions or null.
+ */
+export const getMediaAccessor = function(mainManifest, uri) {
+  let result = null;
+
+  forEachPlaylist(mainManifest, function(media, index, array) {
+    if (media.uri === uri) {
+      result = {
+        get: () => array[index],
+        set: (v) => {
+          array[index] = v;
+        }
+      };
+
+      return true;
+    }
+  });
+
+  return result;
+};
+
+/**
+ * A class to encapsulate all of the functionality for
+ * Dash media playlists. Note that this PlaylistLoader does
+ * not refresh, parse, or have manifest strings. This is because
+ * Dash doesn't really have media playlists. We only use them because:
+ * 1. We want to match our HLS API
+ * 2. Dash does have sub playlists but everything is updated on main.
+ *
+ * @extends PlaylistLoader
+ */
+class DashMediaPlaylistLoader extends PlaylistLoader {
+  /**
+   * Create an instance of this class.
+   *
+   * @param {string} uri
+   *        The uri of the manifest.
+   *
+   * @param {Object} options
+   *        Options that can be used. See base class for
+   *        shared options.
+   *
+   * @param {boolean} options.mainPlaylistLoader
+   *        The main playlist loader this playlist exists on.
+   */
+  constructor(uri, options) {
+    super(uri, options);
+    this.manifest_ = null;
+    this.manifestString_ = null;
+
+    this.sidx_ = null;
+    this.mainPlaylistLoader_ = options.mainPlaylistLoader;
+    this.boundOnMainUpdated_ = () => this.onMainUpdated_();
+
+    this.mainPlaylistLoader_.on('updated', this.boundOnMainUpdated_);
+
+    // turn off event watching from parent
+    this.off('refresh', this.refreshManifest_);
+    this.off('updated', this.setMediaRefreshTimeout_);
+  }
+
+  // noop, as media playlists in dash do not have
+  // a uri to refresh or a manifest string
+  refreshManifest_() {}
+  parseManifest_() {}
+  setMediaRefreshTimeout_() {}
+  clearMediaRefreshTimeout_() {}
+  getMediaRefreshTime_() {}
+  getManifestString_() {}
+
+  /**
+   * A function that is run when main updates, but only
+   * functions if this playlist loader is started. It will
+   * merge it's old manifest with the new one, and update it
+   * with sidx segments if needed.
+   *
+   * @listens {DashMainPlaylistLoader#updated}
+   * @private
+   */
+  onMainUpdated_() {
```

Comment:
This is the main function for this file. It finds the playlist this loader is associated with. Adds sidx segments (if needed) and merges with it's reference of it's old playlist.